Practice Areas/Engineering/RFCs/2019-09-02 Vets-Website CMS Integration.md

```diff
@@ -0,0 +1,85 @@
+1. Since the bad link manifests itself as in error vets-website - a domain maintained by engineers - __the error is not raised to the practice area that manages it__ - content. 
+1. As the CMS content continues to expand under WBC (Web Brand Consolidation), __there will be potentially hundreds of users updating content in the CMS__. This exponentially increases the risk of erroneous content being introduced.
+
+### But doesn't the CMS cache fix this issue?
+When a build in vets-website fails during Jenkins CI, the build will fallback to a CMS "cache" and attempt to rebuild. This model allows us to build the site using the last "good" data pulled from the CMS, essentially allowing us to deploy application code without worrying about grabbing the latest content. This model is complex and not foolproof, as we saw a broken link in both latest CMS and CMS cache [today](http://jenkins.vfs.va.gov/blue/organizations/jenkins/testing%2Fvets-website/detail/master/5213/pipeline), and it also prevents new CMS content from being deployed even though the broken link itself may live in a completely separate area of the site. Since builds fallback to the CMS cache almost exclusively due to broken links, the work proposed by this PR could be an opportunity to reconcile the concept of CMS cache with the latest CMS data, creating a single source of truth, since (spoiler alert!) broken links would no longer be a reason to invalidate CMS data.
+
+## Design
+_Explain the proposed changes in enough detail so that a team member with working knowledge 
+of the project could implement the change themselves. Bonus points if this is easily mappable
+to user stories or specs._
+
+The link validation is the more pressing issue since it occurs more often, but the accessibility tests would need to be addressed as well. I recommend doing this in two pieces.
+
+### Broken-Link Checker
+
+__The broken-link-checker should no longer execute during a website build, but during a later stage of the CI.__ It should execute similarly to the `prearchive` stage of our CI, where the website has already been successfully built, and occur only when the CI is executing on the master branch. The outputted HTML is then processed and evaluated for broken links. We can summarize the result into a "report", with criteria that could include number of pages processed, number of links processed, any broken links found, etc. 
+
+We could also define some build-breaking criteria, where we have reached a certain threshold that our is determined by our build to not be deployable. For example, perhaps 100 broken links is _okay_ (considering the website will eventually be composed of many, many pages), but if there are 100 broken links to the same resource (such as `/health-care`), then this is not an acceptable state for the website to be deployed, because a key resource is missing. In this case, the build or deployment will fail.
```

Comment:
Who would be responsible for determining these "allowable degradation" rules and thresholds?

I'm definitely not a fan of intentionally allowing any degradation, but I think this needs to be a DSVA decision.